src/main/java/bwapi/Game.java

```diff
@@ -1014,23 +1008,23 @@ final public boolean hasPowerPrecise(final Position position, final UnitType uni
         return hasPowerPrecise(position.x, position.y, unitType);
     }
 
-    final public boolean hasPower(final int tileX, final int tileY) {
-        return hasPower(new TilePosition(tileX, tileY));
+    final public boolean hasPower(final TilePosition position) {
+        return hasPower(position.x, position.y);
     }
 
-    final public boolean hasPower(final int tileX, final int tileY, final UnitType unitType) {
-        return hasPower(new TilePosition(tileX, tileY), unitType);
+    final public boolean hasPower(final int tileX, final int tileY) {
+        return hasPower(tileX, tileY, UnitType.None);
     }
 
-    final public boolean hasPower(final TilePosition position) {
-        return hasPower(position.x, position.y, UnitType.None);
+    final public boolean hasPower(final TilePosition position, final UnitType unitType) {
+        return hasPower(position.x, position.y, unitType);
     }
 
-    final public boolean hasPower(final TilePosition position, final UnitType unitType) {
+    final public boolean hasPower(final int tileX, final int tileY, final UnitType unitType) {
         if (unitType.id >= 0 && unitType.id < UnitType.None.id) {
-            return hasPowerPrecise(position.x * 32 + unitType.tileWidth() * 16, position.y * 32 + unitType.tileHeight() * 16, unitType);
+            return hasPowerPrecise(tileX * 32 + unitType.tileWidth() * 16, tileY * 32 + unitType.tileHeight() * 16, unitType);
         }
-        return hasPowerPrecise(position.x * 32, position.y * 32, UnitType.None);
+        return hasPowerPrecise(tileY * 32, tileY * 32, UnitType.None);
```

Comment:
Allowing access to `hasPower[Precise]()` without having to allocate a TilePosition.